malfeasance/wire/malfeasance.go

```diff
@@ -103,7 +103,7 @@ func (e *Proof) EncodeScale(enc *scale.Encoder) (int, error) {
 	var total int
 	{
 		// not compact, as scale spec uses "full" uint8 for enums
-		n, err := scale.EncodeByte(enc, e.Type)
+		n, err := scale.EncodeByte(enc, byte(e.Type))
```

Comment:
It's already a byte, isn't?
```suggestion
		n, err := scale.EncodeByte(enc, e.Type)
```